### PR TITLE
google cloud connection, controllers, service for google login

### PR DESCRIPTION
**Pull request by KashyretsIvanna**
fix: env cross-env
feature: google-login - example feature:
What to check:
-  google cloud connection
- controllers for google connection
- services for google connection
Result of google connection: 
зображення

### CODE REVIEW

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
-@Controller('auth')
+@Controller('/auth')
 export class AuthController {
   constructor(private readonly authService: AuthService) {}
+
+  // eslint-disable-next-line class-methods-use-this
+  @Get('/google')
+  @UseGuards(AuthGuard('google'))
+  async googleAuth() {
+    return 'google auth';
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts]: hmm look useless

File: src/modules/auth/googleauth/google.service.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, VerifyCallback } from 'passport-google-oauth20';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
+  constructor() {
+    super({
+      clientID: `${process.env.GOOGLE_CLIENT_ID}`,
```
Comment (lyudad) [inline src/modules/auth/googleauth/google.service.ts]: if you add something to env, you need to have env.example file in other case rest of the team will have problem with project start

File: src/modules/auth/googleauth/google.service.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      clientSecret: `${process.env.GOOGLE_CLIENT_SECRET}`,
+      callbackURL: `http://${process.env.HOST}:${process.env.PORT}/auth/google/redirect`,
+      scope: ['email', 'profile'],
+    });
+    this.validate = this.validate.bind(this);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
```
Comment (lyudad) [inline src/modules/auth/googleauth/google.service.ts]: its strange create rule and then disable it everywhere

File: src/modules/auth/googleauth/google.service.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  async validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: any,
```
Comment (lyudad) [inline src/modules/auth/googleauth/google.service.ts]: no any. check git checklist!!!!!!!!!!!!!!!!!!!!!!!!!!

File: src/modules/auth/auth.controller.ts (hunk 1 above)
Comment (KashyretsIvanna) [reply src/modules/auth/auth.controller.ts]: fixed

File: src/modules/auth/googleauth/google.service.ts (hunk 2 above)
Comment (KashyretsIvanna) [reply src/modules/auth/googleauth/google.service.ts]: fixed

File: src/modules/auth/googleauth/google.service.ts (hunk 3 above)
Comment (KashyretsIvanna) [reply src/modules/auth/googleauth/google.service.ts]: fixed

File: src/modules/auth/googleauth/google.service.ts (hunk 4 above)
Comment (KashyretsIvanna) [reply src/modules/auth/googleauth/google.service.ts]: fixed

File: README.md [hunk 5]
```diff
@@ -19,12 +20,14 @@ To run app you must create env files in root directory
...
-* **DB_USERNAME** - database username
-* **DB_PASSWORD** - database password
+- **PORT** - port to run app
+- **DB_NAME** - database name
+- **DB_HOST** - database host
+- **DB_PORT** - database port
+- **DB_USERNAME** - database username
+- **DB_PASSWORD** - database password
+- **GOOGLE_CLIENT_ID** - "71848933982-oc9dn519u5k0dp3pp0a8h20maaokdap8.apps.googleusercontent.com"
+- **GOOGLE_CLIENT_SECRET** - "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"
```
Comment (lyudad) [inline README.md]: I think its secret to be a secret if you push it to github it not a secret anymore, never push any secrets to github, its very easy to check the history and get all creds. and please create .env.examples it needed for env variable, not README.md

File: src/modules/auth/googleauth/dto/profile.dto.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+interface Emails {
```
Comment (lyudad) [inline src/modules/auth/googleauth/dto/profile.dto.ts]: you need negotiate in the team about dto naming, as I saw that guys put Dto in the end, like EmailsDto, I like this approach better as it help to understand that data came from client

File: src/modules/auth/googleauth/google.service.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { Injectable } from '@nestjs/common';
+import { Prof } from '@/modules/auth/googleauth/dto/profile.dto';
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
+  constructor() {
+    super({
+      clientID: `${process.env.GOOGLE_CLIENT_ID}`,
+      clientSecret: `${process.env.GOOGLE_CLIENT_SECRET}`,
+      callbackURL: `http://${process.env.HOST}:${process.env.PORT}/auth/google/redirect`,
```
Comment (lyudad) [inline src/modules/auth/googleauth/google.service.ts]: full path should be in the env, today you have HTTP, tomorrow you will have https, you approach is not righ

Comment (KashyretsIvanna) [reply src/modules/auth/googleauth/google.service.ts]: fixed

File: src/modules/auth/googleauth/dto/profile.dto.ts (hunk 6 above)
Comment (KashyretsIvanna) [reply src/modules/auth/googleauth/dto/profile.dto.ts]: fixed

File: src/modules/auth/auth.service.ts [hunk 8]
```diff
@@ -4,14 +4,22 @@ import {
...
+import { OAuth2Client } from 'google-auth-library';
 import { UserService } from '@/modules/user/user.service';
 import CreateUserDto from '@/modules/user/dto/create-user.dto';
 import SignInDto from '@/modules/auth/dto/signIn.dto';
 import TokenDto from '@/modules/auth/dto/token.dto';
 import AuthResponseDto from '@/modules/auth/dto/auth-response.dto';
+import { CredentialDto } from '@/modules/auth/googleauth/dto/credential';
+
+const client = new OAuth2Client(
+  process.env.GOOGLE_CLIENT_ID,
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: talk to Sergi, he uses next tool for env, really great approach, as he start from good code style, we will try to follow the same

Comment (KashyretsIvanna) [reply src/modules/auth/auth.service.ts]: thanks, I talked)

Comment (lyudad) [reply src/modules/auth/auth.service.ts]: I dont see changes

File: src/modules/auth/auth.controller.ts [hunk 9]
```diff
@@ -23,4 +24,9 @@ export class AuthController {
   signIn(@Body() dto: SignInDto) {
     return this.authService.signIn(dto);
   }
+
+  @Post('/google')
+  async googleAuth(@Body() body: CredentialDto) {
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts:29]: where is return type? please check git checklist 1st item

**Pull request merged**